src/ChineseNumbers/ChineseNumbers.cpp

```diff
@@ -0,0 +1,102 @@
+#include "ChineseNumbers.h"
+
+#include <cmath>
+#include <iostream>
+#include <sstream>
+
+#include "StringUtils.h"
+
+static std::string lowerDigits[] = {"〇", "一", "二", "三", "四",
+                                    "五", "六", "七", "八", "九"};
+static std::string upperDigits[] = {"零", "壹", "貳", "參", "肆",
+                                    "伍", "陸", "柒", "捌", "玖"};
+static std::string lowerPlaces[] = {"千", "百", "十", ""};
+static std::string upperPlaces[] = {"仟", "佰", "拾", ""};
+static std::string higherPlaces[] = {"",   "萬", "億", "兆", "京", "垓",
+                                     "秭", "穰", "溝", "澗", "正", "載"};
+
+static std::string convert4Digits(std::string substring,
```

Comment:
Since you just need to read `substring`, its type can just be `const std::string&`.

Reply:
Fixed in 92e46601d055fb57699128d5d2cb8634a1fef339